beacon-chain/rpc/blockfetcher/fetcher.go

```diff
@@ -0,0 +1,101 @@
+package blockfetcher
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/blockchain"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/db"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/interfaces"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/encoding/bytesutil"
+)
+
+// BlockIdParseError represents an error scenario where a block ID could not be parsed.
+type BlockIdParseError struct {
+	message string
+}
+
+// NewBlockIdParseError creates a new error instance.
+func NewBlockIdParseError(reason error) BlockIdParseError {
+	return BlockIdParseError{
+		message: errors.Wrapf(reason, "could not parse block ID").Error(),
+	}
+}
+
+// Error returns the underlying error message.
+func (e BlockIdParseError) Error() string {
+	return e.message
+}
+
+// Fetcher is responsible for retrieving info related with the beacon block.
+type Fetcher interface {
+	Block(ctx context.Context, blockId []byte) (interfaces.ReadOnlySignedBeaconBlock, error)
+}
+
+// BlockProvider is a real implementation of Fetcher.
+type BlockProvider struct {
+	BeaconDB         db.ReadOnlyDatabase
+	ChainInfoFetcher blockchain.ChainInfoFetcher
+}
+
+func (p *BlockProvider) Block(ctx context.Context, blockId []byte) (interfaces.ReadOnlySignedBeaconBlock, error) {
```

Comment:
The block part of the name `blockId` feels redundant. Do you need to name a variable `blockId` when it's the only (non-context) argument to a function named `Block`?

Reply:
changed to `id`